servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/DefaultLoadBalancer.java

```diff
@@ -82,20 +77,20 @@ final class DefaultLoadBalancer<ResolvedAddress, C extends LoadBalancedConnectio
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultLoadBalancer.class);
 
-    @SuppressWarnings("rawtypes")
-    private static final AtomicReferenceFieldUpdater<DefaultLoadBalancer, List> usedHostsUpdater =
-            AtomicReferenceFieldUpdater.newUpdater(DefaultLoadBalancer.class, List.class, "usedHosts");
-
     @SuppressWarnings("rawtypes")
     private static final AtomicLongFieldUpdater<DefaultLoadBalancer> nextResubscribeTimeUpdater =
             AtomicLongFieldUpdater.newUpdater(DefaultLoadBalancer.class, "nextResubscribeTime");
 
     private static final long RESUBSCRIBING = -1L;
 
     private volatile long nextResubscribeTime = RESUBSCRIBING;
+
+    // writes to these fields protected by `executeSequentially` but they can be read from any thread.
     private volatile List<Host<ResolvedAddress, C>> usedHosts = emptyList();
+    private volatile boolean isClosed;
```

Comment:
What is a motivation to introduce additional `volatile` boolean instead of using `ClosedList` like it was before?

Reply:
To delete code and be more explicit. In a follow up [here](https://github.com/apple/servicetalk/pull/2774) I make `isClosed` non-volatile.